packages/fuel-indexer-schema/src/db/tables.rs

```diff
@@ -39,32 +28,34 @@ pub struct SchemaBuilder {
     types: HashSet<String>,
     /// Schema field mapping is namespaced by type name
     fields: HashMap<String, HashMap<String, String>>,
-    primitives: HashSet<String>,
+    parsed_schema: ParsedGraphQLSchema,
+    is_native: bool,
 }
 
 impl SchemaBuilder {
+    /// Create a new `SchemaBuilder`.
     pub fn new(
         namespace: &str,
         identifier: &str,
         version: &str,
         db_type: DbType,
+        is_native: bool,
     ) -> IndexerSchemaDbResult<SchemaBuilder> {
-        let base_ast = match parse_schema(BASE_SCHEMA) {
-            Ok(ast) => ast,
-            Err(e) => return Err(IndexerSchemaDbError::ParseError(e)),
-        };
-        let (primitives, _) = build_schema_objects_set(&base_ast);
+        let parsed_schema =
+            ParsedGraphQLSchema::new(namespace, identifier, is_native, None)?;
 
         Ok(SchemaBuilder {
             db_type,
             namespace: namespace.to_string(),
             identifier: identifier.to_string(),
             version: version.to_string(),
-            primitives,
+            parsed_schema,
+            is_native,
             ..Default::default()
         })
     }
 
+    /// Generate table SQL for each object in the given schema.
     pub fn build(mut self, schema: &str) -> IndexerSchemaDbResult<Self> {
```

Comment:
Here's a thought:

Is there an instance where we'd want to create a `SchemaBuilder` instance in `SchemaManager::new_schema()` and _not_ immediately call `.build()` on that instance? If not, perhaps we can combine the `new()` and `build()` since all the necessary parts for `build()` are covered in the parameters for `new().

Reply:
@deekerno I would like to put a pin in that and save it for the next round of refactor (I think tables/schema still needs one more round of cleanup then we should be in a good place).